apps/sms/services/activity/client.js

```diff
@@ -0,0 +1,62 @@
+/* global EventDispatcher,
+          threads
+*/
+
+(function(exports) {
+  'use strict';
+
+  /**
+   * Name of the service that is responsible for managing activities.
+   * @type {string}
+   */
+  const SERVICE_NAME = 'activity-service';
+
+  /**
+   * Definition of all available private members.
+   * @type {Object}
+   */
+  const priv = Object.freeze({
+    client: Symbol('ActivityClient.client'),
+
+    onActivityRequest: Symbol('ActivityClient.onActivityRequest')
+  });
+
+  var ActivityClient = function() {
+    var client = this[priv.client] = threads.client(SERVICE_NAME);
```

Comment:
I think we also need 2 parts to make our "business" code easier to read/write/maintain.

as for singleton/factory/class I think we clearly want a singleton (be it like we always did or using a factory method)... unless you want one different client class for each activity? I don't think this is really needed (activity handler could be just one function) but this could be